File: 2470-removing-stars-from-a-string/removing-stars-from-a-string.cpp
class Solution {
public:
    string removeStars(string s) {
        string ans = "";
        if(s == "") return ans;
        stack<char> st;
        for(auto i: s){
            if(i == '*'){
                if(st.empty()) continue;
                st.pop();
            }else{
                st.push(i);
            }
        }

        while(!st.empty()){
            ans += st.top();
            st.pop();
        }

        int i = 0;
        int j = ans.size()-1;
        while(i < j){
            swap(ans[i++], ans[j--]);
        }

        return ans;
    }
};

File: 2470-removing-stars-from-a-string/removing-stars-from-a-string.java
class Solution {
    public String removeStars(String s) {
        Deque<Character> st = new ArrayDeque<>();

        for(Character i: s.toCharArray()){
            if(i == '*'){
                if(st.isEmpty()) continue;
                st.pop();
            }else{
                st.push(i);
            }
        }

        String ans = "";

        Deque<Character> nst = new ArrayDeque<>();

        while(!st.isEmpty()){
            nst.push(st.peek());
            st.pop();
        }

        while(!nst.isEmpty()){
            ans += nst.peek();
            nst.pop();
        }

        return ans;
    }
}